### PR TITLE
cve-2016-2107: fix for fips mode

### PR DESCRIPTION
**Pull request by tomato42**
### Description

make test-cve-2016-2107.py interoperable with servers working in FIPS mode

### Motivation and Context

work towards https://github.com/tlsfuzzer/tlsfuzzer/issues/563

### Checklist

- [x] I have read the [CONTRIBUTING.md](https://github.com/tlsfuzzer/tlsfuzzer/blob/master/CONTRIBUTING.md) document and my PR follows [change requirements](https://github.com/tlsfuzzer/tlsfuzzer/blob/master/CONTRIBUTING.md#change-requirements) therein
- [ ] the changes are also reflected in documentation and code comments
- [ ] all new and existing tests pass (see CI results)
- [ ] [test script checklist](https://github.com/tlsfuzzer/tlsfuzzer/wiki/Test-script-checklist) was followed for new scripts
- [ ] new test script added to `tlslite-ng.json` and `tlslite-ng-random-subset.json`
- [x] new and modified scripts were ran against popular TLS implementations:
  - [x] OpenSSL
  - [ ] NSS
  - [ ] GnuTLS
- [ ] required version of tlslite-ng updated in requirements.txt and README.md

- - -
This change is [Reviewable](https://reviewable.io/reviews/tlsfuzzer/tlsfuzzer/938)

### CODE REVIEW

File: scripts/test-cve-2016-2107.py [hunk 1]
```diff
@@ -84,16 +110,47 @@ def main():
...
 
     conversation = Connect(host, port)
     node = conversation
-    # any CBC will do, but the rest of code expects a 128 block cipher
-    ciphers = [CipherSuite.TLS_RSA_WITH_AES_128_CBC_SHA]
+    ext = {ExtensionType.renegotiation_info: None}
+    if ems:
+        ext[ExtensionType.extended_master_secret] = AutoEmptyExtension()
+    if dhe:
+        groups = [GroupName.secp256r1,
```
Comment (neverpanic) [inline scripts/test-cve-2016-2107.py:137]: Do we want to make this work in `FIPS:OSPP` and `FUTURE` by adding `secp384r1`?

File: scripts/test-cve-2016-2107.py [hunk 2]
```diff
@@ -114,13 +171,28 @@ def main():
...
     conversation = Connect(host, port)
     node = conversation
-    # any CBC will do, but the rest of code expects a 128 bit block cipher and
-    # 160 bit HMAC
-    ciphers = [CipherSuite.TLS_RSA_WITH_AES_128_CBC_SHA]
+    ext = {ExtensionType.renegotiation_info: None}
+    if ems:
+        ext[ExtensionType.extended_master_secret] = AutoEmptyExtension()
+    if dhe:
+        groups = [GroupName.secp256r1,
```
Comment (neverpanic) [inline scripts/test-cve-2016-2107.py:178]: Same would apply here.

**Review by neverpanic — APPROVED** [review]
LG2M.

File: scripts/test-cve-2016-2107.py (hunk 1 above)
Comment (tomato42) [reply scripts/test-cve-2016-2107.py:137]: not really, as that extends to signature algorithms too, so we need to say what's good enough, and P-256 should be that

File: scripts/test-cve-2016-2107.py (hunk 2 above)
Comment (tomato42) [reply scripts/test-cve-2016-2107.py:178]: same as above

**Pull request merged**